backend/src/test/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/controller/ElementControllerTest.java

```diff
@@ -32,45 +34,273 @@ class ElementControllerTest {
-        .andExpect(jsonPath("$[3].eid").value("hauptteil-1_art-3"))
-        .andExpect(jsonPath("$[3].type").value("article"))
-        // conclusion
-        .andExpect(jsonPath("$[4].title").value("Schlussteil"))
-        .andExpect(jsonPath("$[4].eid").value("schluss-1"))
-        .andExpect(jsonPath("$[4].type").value("conclusions"));
+      }
+
+      @Test
+      void returns404IfElementNotFoundByEid() throws Exception {
+        // given
+        var norm = NormFixtures.loadFromDisk("NormWithMultipleMods.xml");
+        when(loadNormUseCase.loadNorm(any())).thenReturn(Optional.of(norm));
+
+        // when
+        mockMvc
+            .perform(
+                get("/api/v1/norms/eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1/elements/NONEXISTENT_EID")
+                    .accept(MediaType.TEXT_HTML))
+            // then
+            .andExpect(status().isNotFound());
+      }
+
+      @Test
+      void returnsBadRequestIfAtIsoDateIsInvalid() throws Exception {
+        // Given
+
+        // When / Then
+        mockMvc
+            .perform(
+                get("/api/v1/norms/eli/bund/bgbl-1/1990/s2954/2022-12-19/1/deu/regelungstext-1/elements/hauptteil-1_para-20?atIsoDate=INVALID")
+                    .accept(MediaType.TEXT_HTML))
+            .andExpect(status().isBadRequest());
+      }
+    }
+
+    @Test
+    void returnsHtmlRendering() throws Exception {
+      // given
+      var norm = NormFixtures.loadFromDisk("NormWithMultipleMods.xml");
+      when(loadNormUseCase.loadNorm(any())).thenReturn(Optional.of(norm));
+      var renderedHtml = "renderedHtml";
+      when(transformLegalDocMlToHtmlUseCase.transformLegalDocMlToHtml(any()))
+          .thenReturn(renderedHtml);
+
+      // when
+      mockMvc
+          .perform(
+              get("/api/v1/norms/eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1/elements/hauptteil-1_art-1")
+                  .accept(MediaType.TEXT_HTML))
+          // then
+          .andExpect(status().isOk())
+          .andExpect(content().string(renderedHtml));
+    }
+
+    @Test
+    void returnsJsonWithElementEidTitleAndType() throws Exception {
+      // given
+      var norm = NormFixtures.loadFromDisk("NormWithMultipleMods.xml");
+      when(loadNormUseCase.loadNorm(any())).thenReturn(Optional.of(norm));
+
+      // when
+      mockMvc
+          .perform(
+              get("/api/v1/norms/eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1/elements/hauptteil-1_art-1")
+                  .accept(MediaType.APPLICATION_JSON))
+          // then
+          .andExpect(status().isOk())
+          .andExpect(jsonPath("eid").value("hauptteil-1_art-1"))
+          .andExpect(jsonPath("type").value("article"))
+          .andExpect(jsonPath("title").value("Artikel 1 Änderung des Vereinsgesetzes"));
+    }
+
+    // TODO Hannes: I think we can't test this as we mock the result, so it does not show that the
```

Comment:
@andreasphil I'd like to get your input on this one.

Reply:
Agree 👍 this sounds like an integration test for me